proposals/stdlib/mutablelist-move.md

```diff
@@ -0,0 +1,145 @@
+# Adding moveXXX() Functions to MutableList
+
+It would be helpful to create some additional extension functions for `MutableList` to move items to new indexes, as well as move them up/down, and move one or all items that meet a predicate.
+
+
+# Title
+
+* **Type**: Standard Library API proposal
+* **Author**: Thomas Nield
+* **Contributors**: N/A
+* **Status**: Submitted
+* **Prototype**: Implemented
+
+
+## Summary
+
+It would be helpful to create some additional extension functions for `MutableList` to move items to new indexes, as well as move them up/down, and one or all items that meet a predicate.
+
+These can be implemented using only extension functions on `MutableList`.
+
+## Similar API review
+
+Cannot think of an explicit precendent in another API.
```

Comment:
It looks like `Collections.rotate()` simply shifts the entire `List` up or down for all elements in their current order. It does not move elements relative to each other.  When I specify a condition, I am moving elements with that condition to a certain area in the `List`. 

`subList()` looks only to extract part of the `List` between two indexes. If there is a way to make it work in conjunction with `Collections.rotate()`, I'm not sure what implementation you are envisioning.